src/kio/serial/readers.py

```diff
@@ -127,14 +127,14 @@ def read_raw_bytes(buffer: IO[bytes]) -> bytes | None:
 
 
 def read_legacy_bytes(buffer: IO[bytes]) -> bytes:
-    length = read_int16(buffer)
+    length = read_int32(buffer)
```

Comment:
praise: This looks correct 👍 I think I might have mixed this up due to some internal usage where we interpreted the string type as bytes. As a trivia, I believe the `read_raw_bytes()` function just above is unused and should be removed in favor of these two functions. I'll add a ticket to track this.

Reply:
Do you want this as a separate PR?

Reply:
Not necessarily, I think it's OK either way.

Here's the ticket: https://github.com/Aiven-Open/kio/issues/94, mostly to remember to do the deletion at some point.